pkg/controller/instance/instance_controller.go

```diff
@@ -382,22 +386,40 @@ func GetOperatorVersion(instance *kudov1beta1.Instance, c client.Client) (ov *ku
 	return ov, nil
 }
 
-// paramsMap generates {{ Params.* }} map of keys and values which is later used during template rendering.
-func paramsMap(instance *kudov1beta1.Instance, operatorVersion *kudov1beta1.OperatorVersion) map[string]string {
-	params := make(map[string]string)
-
-	for k, v := range instance.Spec.Parameters {
-		params[k] = v
+func paramValue(v *string, t v1beta1.ParameterValueType) (r interface{}, err error) {
+	switch t {
+	case kudov1beta1.DictValueType:
+		r, err = kudo.YAMLDict(kudo.StringValue(v))
+	case kudov1beta1.ListValueType:
+		r, err = kudo.YAMLList(kudo.StringValue(v))
+	default:
+		r = kudo.StringValue(v)
 	}
 
-	// Merge instance parameter overrides with operator version, if no override exist, use the default one
+	return
+}
+
+// paramsMap generates {{ Params.* }} map of keys and values which is later used during template rendering.
+func paramsMap(instance *kudov1beta1.Instance, operatorVersion *kudov1beta1.OperatorVersion) (map[string]interface{}, error) {
+	params := make(map[string]interface{}, len(operatorVersion.Spec.Parameters))
+
 	for _, param := range operatorVersion.Spec.Parameters {
-		if _, ok := params[param.Name]; !ok {
-			params[param.Name] = kudo.StringValue(param.Default)
+		var err error
+
+		params[param.Name], err = paramValue(param.Default, param.Type)
```

Comment:
It seems weird that we're doing something that might be overridden in a moment.
Why not move this into the `else` branch of the block starting on line 414?

Reply:
Mostly to improve readability; we always set the default value and later overwrite it if there is a parameter value.

Reply:
Hmmm... tbh, I dont' find this very readable, especially with the extra error handling. What about something like this/

```
var p string
if value, ok := instance.Spec.Parameters[param.Name]; ok {
  p = &value
} else {
  p = param.Default
} 

params[param.Name], err = paramValue(p, param.Type)
if err != nil {
  return nil, err
}
```

Also: We might want to move `paramValue` to the convert package? Then it would read `convert.paramValue` which makes it a bit more clear than `paramValue by itself`